File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ComboBox/Features/AutocompletePage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.ComboBox.Features
{
    public partial class AutocompletePage : UserControl
    {
        public AutocompletePage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the SelectedIndexChanged event of the mobjAutoCompleteComboBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjAutoCompleteComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Sets combobox's AutoComplete mode according to selected option
            switch (mobjAutoCompleteComboBox.SelectedIndex)
            { 
                //None
                case 0:
                    this.mobjComboBox.AutoCompleteMode = AutoCompleteMode.None;
                    break;
                //Suggest
                case 1:
                    this.mobjComboBox.AutoCompleteMode = AutoCompleteMode.Suggest;
                    break;
                //Append
                case 2:
                    this.mobjComboBox.AutoCompleteMode = AutoCompleteMode.Append;
                    break;
                //SuggestAppend
                case 3:
                    this.mobjComboBox.AutoCompleteMode = AutoCompleteMode.SuggestAppend;
                    break;
            }
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ComboBox/Features/AutocompletePage.vb
Namespace CompanionKit.Controls.ComboBox.Features

	Public Class AutocompletePage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

        End Sub

        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the mobjAutoCompleteComboBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjAutoCompleteComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles mobjAutoCompleteComboBox.SelectedIndexChanged
            'Sets combobox's AutoComplete mode according to selected option
            Select Case mobjAutoCompleteComboBox.SelectedIndex
                'None
                Case 0
                    Me.mobjComboBox.AutoCompleteMode = AutoCompleteMode.None
                    Exit Select
                    'Suggest
                Case 1
                    Me.mobjComboBox.AutoCompleteMode = AutoCompleteMode.Suggest
                    Exit Select
                    'Append
                Case 2
                    Me.mobjComboBox.AutoCompleteMode = AutoCompleteMode.Append
                    Exit Select
                    'SuggestAppend
                Case 3
                    Me.mobjComboBox.AutoCompleteMode = AutoCompleteMode.SuggestAppend
                    Exit Select
            End Select
        End Sub

	End Class

End Namespace